zerver/migrations/0673_rename_general_chat_to_empty_string_topic.py

```diff
@@ -0,0 +1,114 @@
+                        END)
+                    )::text
+                    FROM JSONB_ARRAY_ELEMENTS(edit_history::jsonb) AS elem
+                )
+                WHERE realm_id = %(realm_id)s
+                AND id > %(lower_id_bound)s AND id <= %(upper_id_bound)s
+                AND (
+                    edit_history::jsonb @> '[{{"prev_topic": "test general chat"}}]' OR
+                    edit_history::jsonb @> '[{{"topic": "test general chat"}}]'
+                );
+            """
+            ).format(table_name=Identifier(message_model._meta.db_table))
+            cursor.execute(
+                query,
+                {
+                    "realm_id": realm_id,
+                    "lower_id_bound": lower_id_bound,
+                    "upper_id_bound": upper_id_bound,
+                },
+            )
+
+        print(f"Processed {upper_id_bound} / {max_id}")
+        lower_id_bound += BATCH_SIZE
+
+
+def rename_general_chat_to_empty_string_topic(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    Message = apps.get_model("zerver", "Message")
+    UserTopic = apps.get_model("zerver", "UserTopic")
+    ArchivedMessage = apps.get_model("zerver", "ArchivedMessage")
+
+    # ScheduledMessage = apps.get_model("zerver", "ScheduledMessage")
+    # Draft = apps.get_model("zerver", "Draft")
+
+    for realm in Realm.objects.all():
+        with transaction.atomic(durable=True):
+            # Uses index "zerver_message_realm_upper_subject"
+            message_queryset = Message.objects.filter(
+                realm=realm, subject__iexact="test general chat"
+            )
+            channel_ids = list(
+                message_queryset.distinct("recipient__type_id").values_list(
+                    "recipient__type_id", flat=True
+                )
+            )
+
+            message_queryset.update(subject="")
+
+            # Uses index "zerver_mutedtopic_stream_topic"
+            UserTopic.objects.filter(
+                stream_id__in=channel_ids, topic_name__iexact="test general chat"
+            ).update(topic_name="")
+
+            ArchivedMessage.objects.filter(realm=realm, subject__iexact="test general chat").update(
+                subject=""
+            )
+            # ScheduledMessage.objects.filter(realm=realm, subject__iexact="test general chat").update(subject="")
+            # Draft.objects.filter(topic__iexact="test general chat").update(topic="")
+
+            for message_model in [Message, ArchivedMessage]:
+                update_edit_history(message_model, realm.id)
+
+
+class Migration(migrations.Migration):
+    atomic = False
+    dependencies = [
+        ("zerver", "0672_fix_attachment_realm"),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            rename_general_chat_to_empty_string_topic,
+            reverse_code=migrations.RunPython.noop,
+            elidable=True,
+        ),
+    ]
```

Comment:
I think this commit should also change `EMPTY_TOPIC_FALLBACK_NAME` to "general chat".

Reply:
Yeah, correct. Made this change.